extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -109,13 +109,17 @@ export async function getPublishDatabaseSettings(project: Project, promptForConn
 	// Add Create New at the top now so it'll show second to top below the suggested name of the current project
 	dbQuickpicks.unshift({ label: `$(add) ${constants.createNew}`, isCreateNew: true });
 
-	// Ensure the project name is an option, either adding it if it doesn't already exist or moving it to the top if it does
-	const projectNameIndex = dbs.findIndex(db => db === project.projectFileName);
+	// if a publish profile was loaded and had a database name, use that instead of the project file name
+	const dbName = publishProfile?.databaseName && publishProfile?.databaseName !== '' ? publishProfile?.databaseName : project.projectFileName;
```

Comment:
Can you simplify this a bit? Do we need the second condition there?

```
const dbName = publishProfile?.databaseName ? publishProfile.databaseName : project.projectFileName;
```

Reply:
thanks for catching that! I had only tried `const dbName = publishProfile?.databaseName ?? project.projectFileName` and added the extra check because databaseName being an empty string wasn't being handled correctly with that one.